orocos_kdl/CMakeLists.txt

```diff
@@ -1,7 +1,7 @@
 #
 # Test CMake version
 #
-CMAKE_MINIMUM_REQUIRED(VERSION 2.6)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.0.2)
```

Comment:
If I understand correctly, something like

```cmake
CMAKE_MINIMUM_REQUIRED(VERSION 2.6)
IF(POLICY CMP0048)
  cmake_policy(SET CMP0048 NEW)
ENDIF()

PROJECT(orocos_kdl)
```

would also work to avoid the author warning and still retains the backwards-compatibility to older CMake versions (e.g. on Ubuntu 14.04). Or

```cmake
SET( KDL_VERSION 1.4.0)
IF(CMAKE_VERSION VERSION_LESS 3.0)
    PROJECT(orocos_kdl)
ELSE()
    cmake_policy(SET CMP0048 NEW)
    PROJECT(orocos_kdl VERSION "${KDL_VERSION}")
ENDIF()
```

On the other hand it might be time to switch to CMake 3.0 and apply some [modern CMake](https://cliutils.gitlab.io/modern-cmake/)...

Reply:
I think all of those would work. How should we proceed? Does `master` branch need to stay compatible with Ubuntu Trusty, or is it appropriate to bump the CMake version so that future PRs can use modern CMake features?

Reply:
> Does master branch need to stay compatible with Ubuntu Trusty, or is it appropriate to bump the CMake version so that future PRs can use modern CMake features?

Modern CMake would not be the only reason for an overhaul of KDL. The API itself could also benefit from some modern C++ features and there is quite some legacy (for example https://github.com/orocos/orocos_kinematics_dynamics/pull/215#issuecomment-588243675).

Unfortunately that is not realistic on our side anytime soon, so we would aim for a backwards-compatible release of version 1.4 for ROS Noetic without the need to maintain separate branches.